From 6e5db5e1f0c70bdacffbaf9060305d179a060829 Mon Sep 17 00:00:00 2001
From: YunYi Yang <yangyunyi2@huawei.com>
Date: Sat, 26 Aug 2023 16:49:36 +0800
Subject: [PATCH 066/256] fix kabi broken due to change of struct
 blk_mq_alloc_data

driver inclusion
category: bugfix
bugzilla: https://gitee.com/src-openeuler/kernel/issues/I8F7ZR

----------------------------------------------------------------------------

Commit ("blk-mq: pass in request/bio flags to queue
mapping") changes the struct blk_mq_alloc_data, so we need to
fix kabi broken problem.

Signed-off-by: YunYi Yang <yangyunyi2@huawei.com>
---
 block/blk-mq.h | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/block/blk-mq.h b/block/blk-mq.h
index 29f88f9caa62..1cbe367d249e 100644
--- a/block/blk-mq.h
+++ b/block/blk-mq.h
@@ -148,11 +148,13 @@ struct blk_mq_alloc_data {
 	struct request_queue *q;
 	blk_mq_req_flags_t flags;
 	unsigned int shallow_depth;
-	unsigned int cmd_flags;
 
 	/* input & output parameter */
 	struct blk_mq_ctx *ctx;
 	struct blk_mq_hw_ctx *hctx;
+#ifndef __GENKSYMS__
+	unsigned int cmd_flags;
+#endif
 };
 
 static inline struct blk_mq_tags *blk_mq_tags_from_data(struct blk_mq_alloc_data *data)
-- 
2.27.0

